src/rules/rule_engine.cpp

```
@@ -286,6 +286,49 @@ struct CompiledRule {
+ * source, test fixtures, and check documentation whose entire purpose is to
+ * contain the patterns being detected. @see docs/designs/rule-engine-config.md
+ */
+static bool line_suppresses_file(re2::StringPiece line, const std::string& rule_id, const std::string& category) {
+  std::string s(line.data(), line.size());
+  size_t pos = s.find("cpm:ignore-file");
+  if (pos == std::string::npos) return false;
+  std::string rest = s.substr(pos + 15);
+  size_t start = rest.find_first_not_of(" \t:,");
+  if (start == std::string::npos) return true;  // bare directive suppresses all
+  rest = rest.substr(start);
+  if (!rule_id.empty() && rest.find(rule_id) != std::string::npos) return true;
```

Clang (14.0.6)

[warning] statement should be inside braces

(readability-braces-around-statements)